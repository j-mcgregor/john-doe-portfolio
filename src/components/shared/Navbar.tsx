import React from 'react'
import { Link, useStaticQuery, graphql } from 'gatsby'
import { NavbarProps } from '../../types/interfaces'
import Img from 'gatsby-image'

const Navbar: React.FC<NavbarProps> = ({ brand }) => {
    const data = useStaticQuery(graphql`
        {
            file(
                sourceInstanceName: { eq: "images" }
                relativePath: { eq: "logo.png" }
            ) {
                childImageSharp {
                    fixed(width: 50, height: 50) {
                        base64
                        width
                        height
                        src
                        srcSet
                    }
                }
            }
        }
    `)

    const logo =
        data && data.file && data.file.childImageSharp.fixed ? (
            <Img fixed={data.file.childImageSharp.fixed} alt="Logo" />
        ) : (
            brand
        )

    return (
        <nav className="navbar">
            <Link to="/" className="brand">
                {logo}
            </Link>
            <ul className="nav-list">
                <li className="nav-item">
                    <Link to="/contact">Contact</Link>
                </li>
            </ul>
        </nav>
    )
}

export default Navbar
