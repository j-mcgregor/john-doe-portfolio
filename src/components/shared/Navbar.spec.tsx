import React from 'react'
import renderer from 'react-test-renderer'
import * as Gatsby from 'gatsby'
import Navbar from './Navbar'

const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery')

describe('Navbar', () => {
    it('should render with a text element if no image found', () => {
        useStaticQuery.mockImplementation(() => ({
            file: null,
        }))
        const brand = <>Test</>

        const nav = renderer.create(<Navbar brand={brand} />)
        const navInstance = nav.toJSON()

        expect(navInstance).not.toBeNull()
        if (navInstance) {
            expect(navInstance.type).toBe('nav')
            expect(navInstance.props).toEqual({ className: 'navbar' })
            expect(navInstance).toMatchSnapshot()
        }
    })

    it('should render with an image if found', () => {
        useStaticQuery.mockImplementation(() => ({
            file: {
                childImageSharp: {
                    fixed: {
                        src: '/static/123/456/logo.png',
                        height: 50,
                        width: 50,
                        srcSet: '',
                        base64: 'data:image/png',
                    },
                },
            },
        }))
        const brand = <>Test</>

        const nav = renderer.create(<Navbar brand={brand} />)
        const navInstance = nav.toJSON()

        expect(navInstance).not.toBeNull()
        if (navInstance) {
            expect(navInstance.type).toBe('nav')
            expect(navInstance.props).toEqual({ className: 'navbar' })
            expect(navInstance).toMatchSnapshot()
        }
    })
})
