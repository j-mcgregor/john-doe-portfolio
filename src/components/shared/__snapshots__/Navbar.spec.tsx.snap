// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render with a text element if no image found 1`] = `
<nav
  className="navbar"
>
  <a
    className="brand"
    href="/"
  >
    Test
  </a>
  <ul
    className="nav-list"
  >
    <li
      className="nav-item"
    >
      <a
        href="/contact"
      >
        Contact
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Navbar should render with an image if found 1`] = `
<nav
  className="navbar"
>
  <a
    className="brand"
    href="/"
  >
    <div
      className=" gatsby-image-wrapper"
      style={
        Object {
          "display": "inline-block",
          "height": 50,
          "overflow": "hidden",
          "position": "relative",
          "width": 50,
        }
      }
    >
      <img
        alt=""
        aria-hidden={true}
        src="data:image/png"
        style={
          Object {
            "height": "100%",
            "left": 0,
            "objectFit": "cover",
            "objectPosition": "center",
            "opacity": 1,
            "position": "absolute",
            "top": 0,
            "transitionDelay": "500ms",
            "width": "100%",
          }
        }
      />
      <picture>
        <source
          srcSet=""
        />
        <img
          alt="Logo"
          height={50}
          loading="lazy"
          onLoad={[Function]}
          src="/static/123/456/logo.png"
          srcSet=""
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 0,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 500ms",
              "width": "100%",
            }
          }
          width={50}
        />
      </picture>
      <noscript
        dangerouslySetInnerHTML={
          Object {
            "__html": "<picture><source srcset=\\"\\" /><img loading=\\"lazy\\" width=\\"50\\" height=\\"50\\" src=\\"/static/123/456/logo.png\\" alt=\\"Logo\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
          }
        }
      />
    </div>
  </a>
  <ul
    className="nav-list"
  >
    <li
      className="nav-item"
    >
      <a
        href="/contact"
      >
        Contact
      </a>
    </li>
  </ul>
</nav>
`;
